docs/src/xds.md

```diff
@@ -36,6 +29,113 @@ Since the range of resources configurable by the xDS API extends that of Quilkin
+   This must be added under an annotation `quilkin.dev/tokens` in the [GameServer][Agones GameServers]'s spec.
+   For example:
+   ```yaml
+   annotations:
+     Sets two tokens for the corresponding endpoint with values 1x7ijy6 and 8gj3v2i respectively.
+     quilkin.dev/tokens: MXg3aWp5Ng==,OGdqM3YyaQ==
+   ```
+
+   > Since an Agones GameServer can have multiple ports exposed, if multiple ports are in
+   > use, the server looks for the port named `default` and picks that as the endpoint's
+   > port (otherwise it picks the first port in the port list).
+
+2. Filter chain is configurable on a per-proxy basis. By default an empty filter chain is used and from there the filter chain can configured using a configMap name `quilkin-config` on the proxy's pod.
+
+As an example, the following runs the server with subcommnad _agones_ against a cluster (using default kubeconfig configuration) where Quilkin pods run in the `quilkin` namespace and game-server pods run in the `gameservers` namespace:
+
+```sh
+quilkin manage --port 18000 agones --config-namespace quilkin --gameservers-namespace gameservers
+```
+
+> A proxy's pod must have a `quilkin.dev/role` key in `quilkin-config` configMap set to the value `proxy` in order for the management server to detect the pod as a proxy and push updates to it.
+
+> Note that currently, the server can only discover resources within a single cluster.
+
+### Filesystem
+
+The file server command is primarily an example and mostly suitable for demo purposes. As a result, some configuration options and features might be missing. This file implementation watches a configuration file on disk and sends updates to proxies whenever that file changes.
+
+It can be started with using subcommnad _file_ as the following:
+```sh
+quilkin manage --port 18000 file --config-file-path config.yaml
+```
+
+After running this command, any proxy that connects to port 18000 will receive updates as configured in `config.yaml` file.
+
+The configuration file schema is:
+```yaml
+# clusters contain a list of clusters.
+# Each entry represents a cluster configuration.
+clusters: [{
+  # Name of the cluster.
+  name: string
+
+  # List of endpoints belonging to the cluster.
+  # Each entry represents an upstream endpoint.
+  endpoints: [{
+    # The endpoint's IP address.
+    ip: int
+    # The endpoint's port.
+    port: int
+    # Opaque metadata that will be the endpoint's metadata.
+    metadata: {}
+  }]
+}]
+
+# filterchain represents the filter chain configuration.
+# It contains a list of filter configurations.
+filterchain: [{
+  # Name of the filter
+  name: string
+
+  # typed_config contains the filter's configuration.
+  typed_config: {
+    # @type must be equivalent to name - the name of the filter.
+    # It is an extra, required field.
+    '@type': string
+    # ...
+    # The rest of the body contains filter specific configuration or
+    # is empty if the filter has no configuration.
+  }
+}]
+```
```

Comment:
I'd remove all of this and just point people at https://googleforgames.github.io/quilkin/main/book/proxy-configuration.html as this is the configuration the provider will be using.

```suggestion
```